test/wallet-http-test.js

```diff
@@ -305,21 +307,22 @@ describe('Wallet HTTP', function() {
 
     const opens = mtx.outputs.filter(output => output.covenant.type === types.OPEN);
     assert.equal(opens.length, 1);
+
+    // reset for next test
+    node.mempool.removeListener('tx', handler);
   });
 
   it('should create an open and not sign the transaction', async () => {
+    let entered = false;
+    const handler = () => entered = true;
+    node.mempool.on('tx', handler);
```

Comment:
It could be possible to use the event handler `once` instead of `on` which would would remove the need to call `removeListener` below. A benefit of using `on` over `once` would be testing that the handler is called exactly once

Reply:
True - but the point of the test is the event never fires!

Reply:
`assert.equal(entered, false);`

So no need for `once` and the event listener has to be removed no matter what

Reply:
Thats what I'm thinking! ...?